packages/mds-metrics/request-handlers.ts

```diff
@@ -201,30 +201,42 @@ export async function getEventCounts(req: MetricsApiRequest, res: GetEventCounts
 
   **Note: unlike the above methods, this method exclusively uses URL query params**
 */
+
 export async function getAll(req: MetricsApiRequest, res: GetAllResponse) {
   const { query } = req
-  const bin_size = getBinSizeFromQuery(query)
+  const bin_size = getBinSize(query.bin_size)
 
   const { start_time, end_time } = parseRelative(query.start || 'today', query.end || 'now')
-  const slices = getTimeBins({
-    bin_size,
-    start_time,
-    end_time
-  })
-  const provider_id = query.provider_id || null
-  const vehicle_type = query.vehicle_type || null
+  const slices = bin_size
+    .map(currBinSize => {
+      return getTimeBins({
+        bin_size: currBinSize,
+        start_time,
+        end_time
+      })
+    })
+    .reduce((prevSlices, currSlices) => {
+      return prevSlices.concat(currSlices)
+    }, [])
+  const provider_id: UUID[] = normalizeToArray<UUID>(query.provider_id)
```

Comment:
It seems super weird to me that `query.provider_id` can either be a single `UUID` that we convert into a list of UUIDs of length 1, or a list of `UUID`s... Should just be `query.provider_ids` and the client should be required to wrap even a single UUID in an array instead of the backend trying to figure out what to do.

Reply:
This is coming from `req.query` ... the way Express works is that provider_id will be a `string` for `?provider_id=foo`, but a `string[]` for `?provider_id=foo&provider_id=bar` .. here in the request handler would be a logical place to normalize the format. This is a pretty common pattern for dealing with query string parameters that may be repeated.

Reply:
Considering Express can handle JSON, it seems much more logical to just have an array of UUIDs though, especially because if we get a `string` we just wrap it up in an array anyways.

Reply:
Could also just do something like `provider_ids[]='foo'?provider_ids[]='bar'` and express should handle that for us to make it always be a `string[]`. This seems to be the most commonly used pattern in JSON APIs nowadays from what I've read.

Reply:
Or alternatively, `provider_ids=['foo', 'bar']`

Reply:
I disagree with putting the onus on the client to format the values as JSON (or an array) when passing data via the query string (request bodies are a different story). Most http libraries will handle the query string parsing and create arrays for properties specified more than once. If you are trying to send a request using a REST client, or curl, you don't necessarily want to have to stringify JSON parameters manually.

Reply:
In that case, this variable we set this to should probably be renamed to `provider_ids` at least.

Reply:
Yup, I'm onboard with that